spec/models/court_data_spec.rb

```diff
@@ -0,0 +1,304 @@
+require 'rails_helper'
+
+RSpec.describe CourtData do
+  subject(:court_data) { described_class.new(claim_id: claim.id) }
+
+  let(:claim) do
+    create(:claim, case_number: 'TEST12345', create_defendant_and_rep_order: false, defendants: claim_defendants)
+  end
+  let(:claim_defendants) { [] }
+
+  before do
+    search_data.each do |search|
+      allow(LAA::Cda::ProsecutionCase).to receive(:search).with(**search[:params]).and_return(search[:response])
+    end
+  end
+
+  describe '#case_number' do
+    subject { court_data.case_number }
+
+    context 'with a matching case number in Common Platform' do
+      let(:search_data) do
+        [
+          {
+            params: { prosecution_case_reference: 'TEST12345' },
+            response: [instance_double(LAA::Cda::ProsecutionCase, case_number: 'TEST12345')]
+          }
+        ]
+      end
+
+      it { is_expected.to match(hash_including(hmcts: 'TEST12345', claim: 'TEST12345')) }
+    end
+
+    context 'with there is no case number in Common Platform and matching defendant' do
```

Comment:
I think the wording of this context could be clearer - could it be something like `when there is no case number in Common Platform but there is a matching defendant`?